src/parser.rs

```diff
@@ -349,54 +355,63 @@ pub fn master_playlist_from_tags(mut tags: Vec<MasterPlaylistTag>) -> MasterPlay
     master_playlist
 }
 
-named!(pub variant_stream_tag<VariantStream>,
-    do_parse!(tag!("#EXT-X-STREAM-INF:") >> attributes: key_value_pairs >>
-           ( VariantStream::from_hashmap(attributes, false)))
-);
+pub fn variant_stream_tag(i: &[u8]) -> IResult<&[u8], VariantStream> {
```

Comment:
I wonder if various of these parsers should be private and not be exported from the crate. They don't seem very useful by themselves. @rutgersc

Reply:
Good point, most of them except the top level ones don't seem worth it to depend on this library for.

Reply:
I'll do that in a follow-up PR then once this is in. I also have some other minor cleanup pending then.

Reply:
I.e. please wait before doing a new release (which needs to be a breaking release FWIW because of the nom version change) and don't do it directly after merging this :)

Reply:
Aye, no problem. Just let me know when all your changes are in, then i'll bump to version 3.0.0 and do the release. Will you  include those cleanups in the follow up PR then?

Reply:
Yes, let's get this one merged for now. Easier to review if it doesn't grow to much :)

Reply:
I've added them now to this PR as I had a few more minutes to clean things up. There's nothing else coming from me for now, and if you want to release this please don't forget to make it `3.0.0` :)

Reply:
I've updated the version to `3.0.0` now already so it's not forgotten.

Reply:
Haha, even better. I see you've been cleaning house, thanks again for all these changes. I'll release the new version shortly!